pytorch_lightning/utilities/parsing.py

```diff
@@ -197,72 +197,69 @@ def __repr__(self):
         return out
 
 
-def lightning_hasattr(model, attribute):
-    """ Special hasattr for lightning. Checks for attribute in model namespace,
-        the old hparams namespace/dict, and the datamodule. """
+def lightning_get_all_attr_holders(model, attribute):
```

Comment:
Why do we need a function to get all holders? This wasn't required before

Reply:
It's required for `setattr`.  I didn't catch it at first, but since `setattr` doesn't return, it sets **all** of the attributes that match, not just one from one holder.  The old `getattr` and `hasattr` update their local value which overwrites the past ones, but `setattr` has no such mechanism.